app/controllers/api_controller/service_requests.rb

```diff
@@ -29,6 +29,17 @@ def service_requests_remove_resource(target, type, id, _data)
       end
     end
 
+    def find_service_requests(id)
+      klass = collection_class(:service_requests)
+      return klass.find(id) if @auth_user_obj.admin?
+      klass.find_by!(:requester => @auth_user_obj, :id => id)
+    end
+
+    def service_requests_search_conditions
+      return {} if @auth_user_obj.admin?
+      {:requester => @auth_user_obj}
+    end
+
```

Comment:
@jrafanie good call! (No pun intended). There is some :sparkles: magic :sparkles: going on here. I was very reluctant to add support for this previously (it would be great if we could pass a strategy in in a more OO way but I don't think that's possible as it stands). I will update the commit to explain a bit more. Thanks!